utils/fork-tree/src/lib.rs

```diff
@@ -351,14 +354,16 @@ where
 		})
 	}
 
-	/// Prune the tree, removing all non-canonical nodes. We find the node in the
-	/// tree that is the deepest ancestor of the given hash and that passes the
-	/// given predicate. If such a node exists, we re-root the tree to this
-	/// node. Otherwise the tree remains unchanged. The given function
-	/// `is_descendent_of` should return `true` if the second hash (target) is a
-	/// descendent of the first hash (base).
+	/// Prune the tree, removing all non-canonical nodes.
```

Comment:
The PR mostly involves the `ForkTree::prune` method.

The rest of the PR are mostly tests